pylint/checkers/base.py

```diff
@@ -1215,6 +1269,30 @@ def open(self):
             for name_type in group.split(':'):
                 self._name_group[name_type] = 'group_%s' % (group,)
 
+        regexps, hints = self._create_naming_rules()
+        self._name_regexps = regexps
+        self._name_hints = hints
+
+    def _create_naming_rules(self):
+        regexps = {}
+        hints = {}
+        naming_style = NAMING_STYLES[self.config.naming_style]
+
+        for name_type in naming_style.known_node_types():
+            regexps[name_type] = naming_style.get_regex(name_type)
+            overriden_value_name = "%s_rgx" % (name_type,)
+            overriden_value = getattr(self.config, overriden_value_name, None)
+            if overriden_value is not None:
+                regexps[name_type] = overriden_value
+
+            hints[name_type] = regexps[name_type].pattern
```

Comment:
Yes, something like that. Just another step of making Pylint more user friendly (another feature similar to #1035 - don't tell me it's wrong, tell me how to fix it).

I'm aware that in general case it's quite hard and will be some kind of heuristic algorithm , but at the end, we can run heuristic result against predefined regex, and if it won't match just do not include suggestion out of message.

Sounds much better than these static hints that are available currently.